diff --git a/kernel_platform/oplus/bazel/oplus_modules_variant.bzl b/kernel_platform/oplus/bazel/oplus_modules_variant.bzl
index e5ae4bae..2df5b54b 100755
--- a/kernel_platform/oplus/bazel/oplus_modules_variant.bzl
+++ b/kernel_platform/oplus/bazel/oplus_modules_variant.bzl
@@ -1,4 +1,4 @@
-bazel_support_target = ["sun", "parrot"]
+bazel_support_target = ["sun"]
 bazel_support_variant = ["perf"]
 OPLUS_FEATURES = ""
 LINUX_KERNEL_VERSION = "6.6"
diff --git a/kernel_platform/oplus/build/oplus_setup.sh b/kernel_platform/oplus/build/oplus_setup.sh
index 27cd5a16..e42b1b0a 100755
--- a/kernel_platform/oplus/build/oplus_setup.sh
+++ b/kernel_platform/oplus/build/oplus_setup.sh
@@ -2,7 +2,7 @@
 
 function init_build_environment() {
 
-    export EXTRA_KBUILD_ARGS="--skip abl"
+    export EXTRA_KBUILD_ARGS="--skip abl ${EXTRA_KBUILD_ARGS:-}"
     export TOPDIR=$(readlink -f ${PWD})
     export ANDROID_BUILD_TOP=${TOPDIR}
     export CHIPSET_COMPANY=QCOM
diff --git a/vendor/oplus/kernel/storage/storage_feature_in_module/common/ufs_oplus_dbg/Makefile b/vendor/oplus/kernel/storage/storage_feature_in_module/common/ufs_oplus_dbg/Makefile
index 7301b88e..f560d819 100755
--- a/vendor/oplus/kernel/storage/storage_feature_in_module/common/ufs_oplus_dbg/Makefile
+++ b/vendor/oplus/kernel/storage/storage_feature_in_module/common/ufs_oplus_dbg/Makefile
@@ -1,2 +1,3 @@
+ccflags-y += -I$(srctree)/drivers/ufs/host/
 ccflags-y += -I$(srctree)/drivers/ufs/core/
 obj-$(CONFIG_OPLUS_UFS_DRIVER)	+= ufs-oplus-dbg.o
diff --git a/vendor/oplus/kernel/storage/storage_feature_in_module/common/ufs_oplus_dbg/ufs-oplus-dbg.c b/vendor/oplus/kernel/storage/storage_feature_in_module/common/ufs_oplus_dbg/ufs-oplus-dbg.c
index e069db4c..1b576e64 100755
--- a/vendor/oplus/kernel/storage/storage_feature_in_module/common/ufs_oplus_dbg/ufs-oplus-dbg.c
+++ b/vendor/oplus/kernel/storage/storage_feature_in_module/common/ufs_oplus_dbg/ufs-oplus-dbg.c
@@ -14,12 +14,12 @@
 
 #include "ufs-oplus-dbg.h"
 #ifdef CONFIG_OPLUS_QCOM_UFS_DRIVER
-#include "../../../../../ufs/host/ufs-qcom.h"
+#include "ufs-qcom.h"
 #endif /* CONFIG_OPLUS_QCOM_UFS_DRIVER */
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 6, 0))
 #ifdef CONFIG_OPLUS_QCOM_UFS_DRIVER
-#include "../../../../../ufs/core/ufshcd-priv.h"
+#include "ufshcd-priv.h"
 #endif /* CONFIG_OPLUS_QCOM_UFS_DRIVER */
 #include "ufshcd-priv.h"
 #else
